pkg/pod/pod.go

```diff
@@ -88,3 +89,22 @@ func RemoveFinalizersFromPod(c client.Client, pl *corev1.PodList) (*RemoveFinali
 		NumRemoved: len(podsWithFinalizersRemoved),
 	}, me.ErrorOrNil()
 }
+
+// Get pod list on a given node matching pod predicate
+func GetPodList(c client.Client, n *corev1.Node, filters []PodPredicate) (*corev1.PodList, error) {
+
+	fieldSelector, err := fields.ParseSelector("spec.nodeName=" + n.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	allPods := &corev1.PodList{}
+	err = c.List(context.TODO(), allPods, &client.ListOptions{
+		FieldSelector: fieldSelector,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return FilterPods(allPods, filters...), nil
```

Comment:
it returns a function here? why cannot return a slice of pods instead?

Reply:
Do you mean this part should be updated to:
```
filteredPods := FilterPods(allPods, filters...), nil
return filteredPods
```
FWIW the old function was also returning the function:
```
return pod.FilterPods(allPods, filters...), nil
```
Functions are returned in other places as seen from
```
$ egrep -Rn "return \w+\(.+\)" pkg/ | wc -l
31
```

Reply:
It seems to be returning the output of that function (`*corev1.PodList`) not the function itself. 
Which seems fine to me given that the function itself only returns one value.